include/onnxruntime/core/session/onnxruntime_ep_c_api.h

```diff
@@ -264,6 +271,53 @@ struct OrtNodeComputeInfo {
   void(ORT_API_CALL* ReleaseState)(_In_ OrtNodeComputeInfo* this_ptr, _Frees_ptr_opt_ void* compute_state);
 };
 
+struct OrtKernelImpl;
+typedef struct OrtKernelImpl OrtKernelImpl;
+
+/**
+ * \brief Contains functions that an OrtEp implements to specify the computation for an operator kernel.
+ * \since Version 1.24.
+ */
+struct OrtKernelImpl {
+  uint32_t ort_version_supported;  ///< Must be initialized to ORT_API_VERSION
+
+  /** \brief Computation function called to execute the kernel on an EP.
+   *
+   * \param[in] this_ptr The OrtKernelImpl instance.
+   * \param[in] context The OrtKernelContext instance that provides access to the inputs and outputs.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.24.
+   */
+  ORT_API2_STATUS(Compute, _In_ OrtKernelImpl* this_ptr, _In_ OrtKernelContext* context);
+
+  /** \brief Called by ORT to release the OrtKernelImpl instance and its resources.
+   *
+   * \param[in] this_ptr The OrtKernelImpl instance.
+   *
+   * \since Version 1.24.
+   */
+  ORT_API_T(void, Release, _In_ OrtKernelImpl* this_ptr);
+};
+
+/** \brief Type definition for a function that creates an OrtKernelImpl instance for an operator kernel.
+ *
+ * \param[in] ctx Unused/reserved for future use.
+ * \param[in] kernel_create_func_state Opaque state initially provided by the EP that registered the kernel.
+ *                                     Refer to OrtEpApi::KernelRegistry_AddKernel(). May be null.
+ * \param[in] info The OrtKernelInfo instance that provides access to the kernel's input and output characteristics.
+ * \param[out] kernel_out Output parameter set to the new OrtKernelImpl instance.
+ *
+ * \snippet{doc} snippets.dox OrtStatus Return Value
+ *
+ * \since Version 1.24.
+ */
+typedef OrtStatus*(ORT_API_CALL* OrtKernelCreateFunc)(_In_ OrtKernelCreateContext* ctx,  // unused/reserved as of 1.24
```

Comment:
We should probably remove this parameter. It is a stand-in for the `FuncManager` parameter in the analogous `KernelCreateFn` used by provider-bridge EPs:
https://github.com/microsoft/onnxruntime/blob/8af9f580860b2bd4ecd3df242ad6e1650e8b9066/include/onnxruntime/core/framework/op_kernel.h#L143

It doesn't look like any EPs in the ORT code base use the `FuncManager` parameter at all, but I kept it here (with a more generic name) just in case we find a use for it in the future. Would appreciate opinions.